framework/src/main/java/org/apache/drill/test/framework/Utils.java

```diff
@@ -99,24 +109,67 @@
       }
     }
     sqlNullabilities = ImmutableMap.copyOf(nullabilityMap);
-    
-    // read configuration file
-    final Map<String, String> properties = Maps.newHashMap();
-    final File overrideFile = new File(CWD + "/conf/" + DRILL_TEST_CONFIG);
-    final ResourceBundle bundle;
-    if (overrideFile.exists() && !overrideFile.isDirectory()) {
+
+    client = getHTTPClientInstance();
+    if (DrillTestDefaults.AUTHENTICATION_MECHANISM.equals("PLAIN")) {
+      authenticateHTTPClient();
+    }
+  }
+
+  /**
+   * Create a HttpClients instance.
+   * @return Returns a default HttpClient instance or a custom instance if SSL is enabled
+   */
+  public static HttpClient getHTTPClientInstance() {
+    HttpClient client;
+    if (DrillTestDefaults.HTTPS_ENABLED) {
+      protocol = "https://";
+
+      Assert.assertFalse("Truststore location not provided", DrillTestDefaults.TRUSTSTORE_PATH.isEmpty());
+      Assert.assertFalse("Truststore password not provided", DrillTestDefaults.TRUSTSTORE_PASSWORD.isEmpty());
+
+      final SSLConnectionSocketFactory socketFactory;
+
       try {
-        bundle = new PropertyResourceBundle(new FileInputStream(overrideFile));
-      } catch (IOException e) {
-        throw new RuntimeException("Error reading configuration file " + overrideFile.getPath(), e);
+        final KeyStore keyStore = KeyStore.getInstance("JKS");
+        try (final InputStream is = new FileInputStream(DrillTestDefaults.TRUSTSTORE_PATH)) {
+          keyStore.load(is, DrillTestDefaults.TRUSTSTORE_PASSWORD.toCharArray());
+        }
+
+        final TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+        tmf.init(keyStore);
+
+        final SSLContext sc = SSLContext.getInstance("TLS");
+        sc.init(null, tmf.getTrustManagers(), new java.security.SecureRandom());
+        socketFactory = new SSLConnectionSocketFactory(sc, new MyHostNameVerifier());
+      } catch (GeneralSecurityException | IOException ex) {
+        LOG.info("Problem validating ssl truststore");
+        throw new RuntimeException(ex);
       }
+      client = HttpClients.custom().setSSLSocketFactory(socketFactory).build();
     } else {
-      bundle = ResourceBundle.getBundle(DRILL_TEST_CONFIG);
+      client = HttpClients.createDefault();
     }
-    for (final String key : bundle.keySet()) {
-      properties.put(key.trim(), bundle.getString(key).trim());
+    return client;
+  }
+
+  /**
+   * Rest server authentication when plain authentication is enabled
+   */
+  public static void authenticateHTTPClient() {
+    HttpPost post = new HttpPost(protocol + DrillTestDefaults.DRILL_STORAGE_PLUGIN_SERVER + ":8047/j_security_check");
```

Comment:
Is there a way to make the port number configurable? Perhaps introduce a new constant in DrillTestDefaults? That can be overridden in drillTestConfig.properties

Reply:
can be done.